src/pages/components/text-inputs.mdx

```diff
@@ -0,0 +1,199 @@
+
+import Icons from '../../examples/text-inputs/MediaInput.tsx';
+import IconsCode from '!!raw-loader!../../examples/text-inputs/MediaInput.tsx';
+
+<CodeExample code={IconsCode}>
+  <Icons />
+</CodeExample>
+
+### FauxInput
+
+A FauxInput makes non-inputs look like inputs.
+
+import FauxInput from '../../examples/text-inputs/FauxInput.tsx';
+import FauxInputCode from '!!raw-loader!../../examples/text-inputs/FauxInput.tsx';
+
+<CodeExample code={FauxInputCode}>
+  <FauxInput />
+</CodeExample>
+
+### Regular weight label
+
+Text input labels are emphasized by default. They can be regular weight using the `isRegular`
+property.
+
+Regular labels are not recommended for text inputs, as they flatten visual hierarchy making
+scanning more difficult.
+
+import RegularWeightLabel from '../../examples/text-inputs/RegularWeightLabel.tsx';
+import RegularWeightLabelCode from '!!raw-loader!../../examples/text-inputs/RegularWeightLabel.tsx';
+
+<CodeExample code={RegularWeightLabelCode}>
+  <RegularWeightLabel />
+</CodeExample>
+
+### Hint
+
+Hint adds additional context to the text Input’s label.
+
+import Hint from '../../examples/text-inputs/Hint.tsx';
+import HintCode from '!!raw-loader!../../examples/text-inputs/Hint.tsx';
+
+<CodeExample code={HintCode}>
+  <Hint />
+</CodeExample>
+
+### Size
+
+A text Input can be default or compact.
+
+import Size from '../../examples/text-inputs/Size.tsx';
+import SizeCode from '!!raw-loader!../../examples/text-inputs/Size.tsx';
+
+<CodeExample code={SizeCode}>
+  <Size />
+</CodeExample>
+
+### Bare
+
+Input borders and padding can be removed using the `isBare` property. Use an
+[aria-label](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-label_attribute)
+to label the input.
+
+import Bare from '../../examples/text-inputs/Bare.tsx';
+import BareCode from '!!raw-loader!../../examples/text-inputs/Bare.tsx';
+
+<CodeExample code={BareCode}>
+  <Bare />
+</CodeExample>
+
+### Focus inset
+
+Box-shadow focus styling can be inset using the `focusInset` property.
+
+This keeps the focus contained within the input area when it is flush with another component.
+
+import FocusInset from '../../examples/text-inputs/FocusInset.tsx';
+import FocusInsetCode from '!!raw-loader!../../examples/text-inputs/FocusInset.tsx';
+
+<CodeExample code={FocusInsetCode}>
+  <FocusInset />
+</CodeExample>
+
+### Placeholder
+
+Placeholder text gives an example of content format. See [Placeholder text](#TODO) standard for
+best practices.
+
+import Placeholder from '../../examples/text-inputs/Placeholder.tsx';
+import PlaceholderCode from '!!raw-loader!../../examples/text-inputs/Placeholder.tsx';
+
+<CodeExample code={PlaceholderCode}>
+  <Placeholder />
+</CodeExample>
+
+### Validation
+
+Success, warning, or danger validation messages are shown using the Message component with the text
+Input.
+
+import Validation from '../../examples/text-inputs/Validation.tsx';
```

Comment:
![image](https://user-images.githubusercontent.com/1230973/84432178-53ca9c80-abe1-11ea-8764-860e24c960b0.png)

The responsive resizing on this example feels off. Could we do something more like what we see on menus/selects? 

![image](https://user-images.githubusercontent.com/1230973/84432292-76f54c00-abe1-11ea-997e-6efddcaba47b.png)

(should probably increase the vertical margin on those examples to give each field more breathing room as well.)

Reply:
Vertical margining with the current grid isn't great. Ideally there would be no margin when side-by-side, and then only interior margining when the elements are stacked.

Reply:
> The responsive resizing on this example feels off. Could we do something more like what we see on menus/selects?

@allisonacs yeah it does feel off, I've made some adjustments. Could you let me know what you think? For small screen sizes the inputs should be stacked now.